### PR TITLE
Doctor's Delight changes

### PR DESCRIPTION
**Pull request by ghost**
- Doctor's Delight now requires cryoxadone instead of omnizine
- Doctor's Delight now has its old effects, restoring half a point of brute, burn, toxin, and oxygen damage per tick
- Doctor's Delight now drains hunger unless the imbiber is a Medical Doctor

As it stands, I have _never_ seen Doctor's Delight made after these changes. It's a pointless cocktail, requiring a rare ingredient (omnizine) and having very little effects compared to something like Neurotoxin or Beepsky Smash that's far easier to acquire and far more powerful.

### CODE REVIEW

**Comment by Iamgoofball** [comment]
@phil235

**Comment by Xhuis** [comment]
I changed the way the drink works. As a downside of the ability to constantly heal, it also drains your hunger quite quickly and you will become hungry if you drink too much. This is not the case for Medical Doctors, hence the name "Doctor's Delight".

**Comment by PKPenguin321** [comment]
i like it

**Comment by Iamgoofball** [comment]
I'm okay with this now. :+1:

**Comment by GrayRachnid** [comment]
Might want to make it work for the CMO too.

**Comment by Fox-McCloud** [comment]
The return of brainless heal-all retardo-chem. :-1:

**Comment by Iamgoofball** [comment]
@Fox-McCloud he gave it a downside this time

and for once the other healing chems are actually superior to using this

**Comment by Fox-McCloud** [comment]
A -1 nutrition drain in exchange for healing 2 of everything? Just throw in a teensy bit of sugar and this is inconsequential.

**Comment by Iamgoofball** [comment]
@Xhuis crank up the nutrition drain a tad then, make sure I can't stack it with sugar

**Comment by Fox-McCloud** [comment]
sugar gives +4, DD gives +1 (ergo the nutrition drain is really only -1)....so it's going to have to drain a whopping 7 nutrition for it to not be abused in conjunction with sugar.

I recommend setting the nutrition var to 0 for DD---still, that's a -5 whopping nutrition to not be able to abuse it with sugar.

maybe all junk food should just be "carbs", instead and have sugar's current nutritional stats and values and "sugar" is just purely an ultra-low end combat chem for reducing stuns. *shrug*

**Comment by Dorsisdwarf** [comment]
Hey maybe hes doing it because he thinks that your shittily-ported chem system removing a meds because "It can be drunk and on goofballstation all medicine is stuck to your forehead" was retarded?

**Comment by Fox-McCloud** [comment]
Because clearly a pants on head retarded cure-all that's just handed out like candy for any ailment is a good and balanced chem that should be celebrated.

I thought the consensus was pushing medical doctors more in a direction of doing different things and being more involved rather than just "slap cure all on person, rinse, wash, repeat".

**Comment by Core0verload** [comment]
:+1: for bringing DD back.

File: code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm [hunk 1]
```diff
@@ -392,9 +392,19 @@
...
+	description = "A gulp a day keeps the Medibot away! A mixture of juices that heals most damage types fairly quickly at the cost of hunger."
 	color = "#FF8CFF" // rgb: 255, 140, 255
 
+/datum/reagent/consumable/doctor_delight/on_mob_life(mob/living/M)
+	M.adjustBruteLoss(-2)
+	M.adjustFireLoss(-2)
+	M.adjustToxLoss(-2)
+	M.adjustOxyLoss(-2)
+	if(M.nutrition && (M.nutrition - 2 > 0))
+		if(!(M.mind && M.mind.assigned_role == "Medical Doctor")) //Drains the nutrition of the holder. Not medical doctors though, since it's the Doctor's Delight!
```
Comment (GunHog) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm]: What about CMO? They are also medical doctors.

Comment (Xhuis) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm]: But it's not in their job name :^)

**Comment by phil235** [comment]
This is terribly OP, especially for doctors. :-1: (Although I like the idea of certain medicine chem making you hungry to give food more use)

**Comment by Xhuis** [comment]
I nerfed the cocktail's new effects. It now heals half a point of each of the types, rather than two. Nutrition drain stays the same, but the drink still isn't useless.

File: code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm [hunk 2]
```diff
@@ -392,9 +392,19 @@
...
+	description = "A gulp a day keeps the Medibot away! A mixture of juices that heals most damage types fairly quickly at the cost of hunger."
 	color = "#FF8CFF" // rgb: 255, 140, 255
 
+/datum/reagent/consumable/doctor_delight/on_mob_life(mob/living/M)
+	M.adjustBruteLoss(-0.5)
+	M.adjustFireLoss(-0.5)
+	M.adjustToxLoss(-0.5)
+	M.adjustOxyLoss(-0.5)
+	if(M.nutrition && (M.nutrition - 2 > 0))
+		if(!(M.mind && M.mind.assigned_role == "Medical Doctor")) //Drains the nutrition of the holder. Not medical doctors though, since it's the Doctor's Delight!
```
Comment (xxalpha) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm]: snowflek

**Comment by xxalpha** [comment]
Why should it matter that doctors are the ones administering the chem?

**Comment by Xhuis** [comment]
Because it's the doctor's delight, honk

**Comment by ExcessiveUseOfCobblestone** [comment]
For thematic purpose of cryoxadone healing at low temperatures, maybe change cream to Ice?

**Comment by Xhuis** [comment]
All the ingredients other than cryoxadone each heal a small amount of each damage type. Doctor's Delight is made to be thematically fitting for that fact.

**Comment by PKPenguin321** [comment]
is there an OD limit because if you got weight even virus and loaded up on 1,000u of this then yeah

File: code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Drink-Reagents/Drinks.dm]: the attitude of "role check = snowflake" is retarded and so are you

**Comment by Lumipharon** [comment]
Why are people saying this is somehow OP (requires shit from chemists and bar), while chemists on their own can make patches that in one click heal far more then sitting there for 10 seconds drinking DD can, only instantly?

**Comment by Dorsisdwarf** [comment]
Because you can drink it, and that makes everything OP in the world of goonchem.

**Comment by Durkel** [comment]
This is probably going to be rarely if ever made to be honest here, and besides the novelty of having DD it just bloats up the chem list we have. There's no point in making something if no one is ever going to use it.

**Comment by Xhuis** [comment]
This wasn't added, it was just repurposed. DD has been in the game longer than I've been around.

**Comment by Xhuis** [comment]
Any issue with merging this?

**Comment by Razharas** [comment]
Im against special effects for roles

**Comment by Iamgoofball** [comment]
Why razharas

**Comment by Xhuis** [comment]
@Razharas Banana juice heals clowns. Nothing heals mimes. I figured that since they already existed, there was no harm in it. If you _really_ feel like this shouldn't be the case, I'll remove it.

**Comment by Lumipharon** [comment]
>singlu effected by engineers, CE and clown
>mime and clown have unique healing items
>captain has magic immunity to cult or whatever the fuck

**Comment by Razharas** [comment]
This is all bad game design and you should feel bad about it

**Comment by GunHog** [comment]
> This is all bad game design and you should feel bad about it

##### Merges anyway

Based Razharas

**Pull request merged**